passenger-service/src/main/java/by/ikrotsyuk/bsuir/passengerservice/service/impl/PassengerServiceImpl.java

```diff
@@ -0,0 +1,103 @@
+package by.ikrotsyuk.bsuir.passengerservice.service.impl;
+
+import by.ikrotsyuk.bsuir.passengerservice.dto.PassengerRequestDTO;
+import by.ikrotsyuk.bsuir.passengerservice.dto.PassengerResponseDTO;
+import by.ikrotsyuk.bsuir.passengerservice.entity.PassengerEntity;
+import by.ikrotsyuk.bsuir.passengerservice.exception.exceptions.PassengerNotFoundByEmailException;
+import by.ikrotsyuk.bsuir.passengerservice.exception.exceptions.PassengerNotFoundByIdException;
+import by.ikrotsyuk.bsuir.passengerservice.exception.exceptions.PassengerWithSameEmailAlreadyExistsException;
+import by.ikrotsyuk.bsuir.passengerservice.mapper.PassengerMapper;
+import by.ikrotsyuk.bsuir.passengerservice.repository.PassengerRepository;
+import by.ikrotsyuk.bsuir.passengerservice.service.PassengerService;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class PassengerServiceImpl implements PassengerService {
+    private final PassengerRepository passengerRepository;
+    private final PassengerMapper passengerMapper;
+
+    @Override
+    public PassengerResponseDTO getPassengerById(Long id){
+        Optional<PassengerEntity> optionalPassengerEntity = passengerRepository.findById(id);
+        return optionalPassengerEntity.map(passengerMapper::toDTO).orElseThrow(() -> new PassengerNotFoundByIdException(id));
+    }
+
+    @Override
+    public Double getPassengerRatingById(Long id){
+        return this.getPassengerById(id).getRating();
+    }
+
+    /**
+     * активирует смену email в keycloak
+     */
+    @Override
+    @Transactional
+    public PassengerResponseDTO editPassengerProfile(Long id, PassengerRequestDTO passengerRequestDTO) {
+        Optional<PassengerEntity> optionalPassengerEntity = passengerRepository.findById(id);
```

Comment:
```suggestion
        Optional<PassengerEntity> optionalPassengerEntity = passengerRepository.findById(id).orElseThrow();
```
тогда твой if не нужен